internal/bminventory/inventory.go

```diff
@@ -2289,6 +2291,160 @@ func (b *bareMetalInventory) DownloadHostLogs(ctx context.Context, params instal
+	}
+
+	manifests, err := b.listManifests(ctx, cluster.ID, "manifests")
+	if err != nil {
+		return installer.NewListClusterManifestsInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	}
+	files, err := b.listManifests(ctx, cluster.ID, "openshift")
+	if err != nil {
+		return installer.NewListClusterManifestsInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	}
+	manifests = append(manifests, files...)
+	return installer.NewListClusterManifestsOK().WithPayload(manifests)
+}
+
+func (b *bareMetalInventory) listManifests(ctx context.Context, clusterID *strfmt.UUID, prefix string) (models.ListManifests, error) {
+	id := string(*clusterID)
+	objectName := filepath.Join(id, prefix)
+	files, err := b.objectHandler.ListObjects(ctx, &objectName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read %s from s3", objectName)
+	}
+	manifests := models.ListManifests{}
+	filePrefix := id + string(os.PathSeparator)
+	for _, file := range files {
+		manifests = append(manifests, &models.Manifest{FileName: strings.TrimPrefix(file, filePrefix)})
+	}
+	return manifests, nil
+}
+
+func (b *bareMetalInventory) DeleteClusterManifest(ctx context.Context, params installer.DeleteClusterManifestParams) middleware.Responder {
+	log := logutil.FromContext(ctx, b.log)
+	fileName := filepath.Join(*params.Folder, params.FileName)
+	log.Infof("Deleting manifest %s from cluster %s", fileName, params.ClusterID)
+
+	var cluster common.Cluster
+	if err := b.db.First(&cluster, identity.AddUserFilter(ctx, "id = ?"), params.ClusterID).Error; err != nil {
+		log.WithError(err).Errorf("failed to get cluster %s", params.ClusterID)
+		return common.NewApiError(http.StatusNotFound, err)
+	}
+
+	objectName := filepath.Join(string(*cluster.ID), fileName)
+	exists, err := b.objectHandler.DoesObjectExist(ctx, objectName)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to delete cluster manifest %s from s3", objectName)
+		return installer.NewDeleteClusterManifestInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, err))
+	}
+
+	if !exists {
+		msg := fmt.Sprintf("Cluster manifest %s doesn't exists in cluster %s", fileName, cluster.ID)
+		log.Warn(msg)
+		return installer.NewDeleteClusterManifestNotFound().WithPayload(common.GenerateError(http.StatusNotFound, errors.New(msg)))
+	}
+
+	if err := b.objectHandler.DeleteObject(ctx, objectName); err != nil {
+		return installer.NewDeleteClusterManifestInternalServerError().
+			WithPayload(common.GenerateError(http.StatusInternalServerError, fmt.Errorf("failed to delete %s from s3", objectName)))
+	}
+
+	log.Infof("Done deleting cluster manifest %s for cluster %s", fileName, cluster.ID)
+	return installer.NewDeleteClusterManifestOK()
+}
+
+func (b *bareMetalInventory) DownloadClusterManifest(ctx context.Context, params installer.DownloadClusterManifestParams) middleware.Responder {
```

Comment:
We have an issue where the UI can't download files via authenticated APIs when in the context of OCM.  This is why we implemented the GetPresignedForClusterFiles API.  I suggest using that API for this context as well: add "manifest" to its file_name parameter enum, and then add a new optional parameter called "additional_name" where the UI would provide "folder_name/manifest_name" as a single string.

@tsorya also needs the same functionality - be aware of each other's work

Reply:
@avishayt  How does this API being tested ? Trying to test it with:
```
curl -X GET ${CLUSTER_API_URL}/downloads/files-presigned?file_name=manifests -H "accept: application/json"
```

Fails with 
```
{"code":"400","href":"","id":400,"kind":"Error","reason":"Failed to generate presigned URL: invalid backend"}
```